### PR TITLE
Player enemy sprites

### PR DESCRIPTION
**Pull request by HusseinAljanaby**
8-directional enemy and player sprites with attack animations. The player attack animations are also 8-directional. Bullets are animated and follow the path of velocity.

### CODE REVIEW

File: Game/Entities/Player.cs [hunk 1]
```diff
@@ -31,6 +52,132 @@ public override void _PhysicsProcess(double delta)
...
+			else 
+			{
+				if (animPlayer.CurrentAnimation != "Die" && animPlayer.IsPlaying() == false) 
+				{
+					animPlayer.Play("Idle Down");
+				}
+			}
+
+		}
+
```
Comment (jamesp6784) [inline Game/Entities/Player.cs:180]: Improve this logic as discussed, maybe with AnimationTree?

File: Menus/Settings/Settings.cs [hunk 2]
```diff
@@ -190,6 +190,12 @@ private void _on_diffculty_option_item_selected(int index)
 		}
 	}
 
+	// controls option function
+	private void _on_controls_option_item_selected(int index)
+	{
+		string selectedOption = ((OptionButton)GetNode("settings/gameplay_settings/controls_option")).GetItemText(index);
```
Comment (jamesp6784) [inline Menus/Settings/Settings.cs:196]: Ideally we wouldn't be using GetNode, we should be using [Export] properties instead so we can set them in the editor

File: Media/Sprites/PlaCDD7.tmp
Comment (jamesp6784) [inline Media/Sprites/PlaCDD7.tmp:1]: Accidental?

File: Game/Entities/Enemy.cs [hunk 3]
```diff
@@ -24,10 +26,12 @@ public enum FiringStyle
 
 	public override void _Ready()
 	{
+		animPlayer = GetNode<AnimationPlayer>("AnimationPlayer");
```
Comment (jamesp6784) [inline Game/Entities/Enemy.cs:29]: Usage of GetNode

File: Game/Entities/Player.cs [hunk 4]
```diff
@@ -13,6 +16,24 @@ public partial class Player : CharacterBody2D
...
+	private string[] animActions = {"Die", 
+	"Attack Up", "Attack Up-Right",
+	 "Attack Right", "Attack Down-Right",
+	  "Attack Down", "Attack Down-Left",
+	   "Attack Left", "Attack Up-Left"};
+
+
+    public override void _Ready()
+    {
+        animPlayer = GetNode<AnimationPlayer>("AnimationPlayer");
```
Comment (jamesp6784) [inline Game/Entities/Player.cs:32]: Usage of GetNode

File: Game/Entities/Projectile.cs [hunk 5]
```diff
@@ -5,7 +5,15 @@ public partial class Projectile : Area2D
 	public float speed = 30f;
 	public float duration = 40;
 
-	public override void _PhysicsProcess(double delta)
+	// Private variable to improve performance
+	private Sprite2D sprite;
+
+    public override void _Ready()
+    {
+        sprite = GetNode<Sprite2D>("Sprite");
```
Comment (jamesp6784) [inline Game/Entities/Projectile.cs:13]: Usage of GetNode

File: Game/Entities/Projectile.cs [hunk 6]
```diff
@@ -14,6 +22,9 @@ public override void _PhysicsProcess(double delta)
 		{
 			QueueFree();
 		}
+
+		// Set the rotation of the sprite to match the direction of the projectile
+		sprite.Rotation = velocity.Angle();
```
Comment (jamesp6784) [inline Game/Entities/Projectile.cs:27]: This is fine for now but ideally I think we would be rotating the projectile itself (and changing our projectile movement logic to fit)

Comment (HusseinAljanaby) [reply Game/Entities/Projectile.cs:27]: Fixing this requires using a rotation pivot, we can discuss if we want to add that!

File: Media/Sprites/PlaCDD7.tmp
Comment (HusseinAljanaby) [reply Media/Sprites/PlaCDD7.tmp:1]: Yes I think so

File: Game/Entities/Projectile.cs (hunk 6 above)
Comment (jamesp6784) [reply Game/Entities/Projectile.cs:27]: Probably just leave for now, we can sort it out later when we do more projectile stuff

File: Game/Entities/Player.cs (hunk 1 above)
Comment (HusseinAljanaby) [reply Game/Entities/Player.cs:180]: Fixed with some angle integer math and renaming animations with indicies.

File: Menus/Settings/Settings.cs (hunk 2 above)
Comment (HusseinAljanaby) [reply Menus/Settings/Settings.cs:196]: I don't believe this is my area, might've accidentally changed something

**Pull request closed without merging**